agent/versioning-publishing.md

```diff
@@ -0,0 +1,62 @@
+# Hypertrace Agents versioning and publishing
+
+This document describes versioning and publishing requirements for Hypertrace agents.
+All Hypertrace agents follow these requirements:
+
+* Semantic versioning for all released artifacts.
+* Release every merge commit into the main or maintenance branch as `"snapshot"`.
+* Maintenance branches follow naming convention `<major>.<minor>.x` e.g. `1.0.x`.
+
+#### Semantic versioning
+
+Each released version strictly follows [semantic versioning](https://semver.org/).
+The released version must be tagged in the source code: e.g. `1.0.0` or `v1.0.0` for languages
+with this convention (e.g. Golang).
+
+#### Snapshot release for every merge commit
+
+Each commit merged into the `main` or maintenance branch (e.g. `release-1.1` for `1.1.x`) should be
+published. The version of this release must include commit `SHA` and should include the last released
+semantic version and `snapshot` string to denote that the release should not be used in production environment.
```

Comment:
Sorry for being late. `snapshot` is a very `Java` term. I think we should go for something more generic like `prerelease` or `alpha` (tho I would vote for `prerelease`)

Reply:
+1 on prerelease